include/swift/Runtime/Config.h

```diff
@@ -147,7 +147,8 @@
 #  define SWIFT_CLASS_IS_SWIFT_MASK 1ULL
 
 // Builds for Swift-in-the-OS always use 2.
-# elif SWIFT_BNI_OS_BUILD
+// Compatibility libraries always build as if they're in the OS.
+# elif SWIFT_BNI_OS_BUILD || SWIFT_COMPATIBILITY_LIBRARY
```

Comment:
I'm guessing this is fixing a linker error on `_swift_classIsSwiftMask`. This value for `SWIFT_CLASS_IS_SWIFT_MASK` won't be correct in the 5.0 compatibility library when running on older OSes. Does that actually impact anything? If not, is there some way we can make sure this value doesn't get used there?

Reply:
As we discussed, the reference here ended up coming via `ClassMetadata::getTypeContextDescriptor`, which looks directly at the bit appropriate for the current runtime. I'll change the override code to use the public `swift_getTypeContextDescriptor` entry point from the host runtime instead.